backend/routes/follows.js

```diff
@@ -4,56 +4,113 @@ import { Stock } from "../models/stock.js";
 
 const router = express.Router();
 
+const checkSession = (req, res, next) => {
+    const user = req.session.user;
+    if (!user) {
```

Comment:
Better to use explicit checks for null and undefined than to rely on coercion of objects to booleans.